### PR TITLE
Add 2023 mwe starter image example

### PR DESCRIPTION
**Pull request by M1chaelM**
## Purpose

This PR adds a minimal working example called `vrx_2023_starter`, analogous to the 2022 image. It's purpose is to give an example image in which the user builds a ROS 2 workspace.

As discussed in [VRX issue 719](https://github.com/osrf/vrx/issues/719), this will also make it easier to update the tutorial.

## To test
At least two things need to be tested. 
* First, check that the instructions work and the image builds and runs.
* Second, make sure images built using this process can be scored in the competition

### Build and Run
* Follow the instructions in the [README file](https://github.com/osrf/vrx-docker/blob/72effaa80cd5ab63240d7c56cfb64a23a369dfb2/mwes/2023/vrx_2023_starter/README.md) to build and run the example image.

### Check scoring
To make sure the resulting image is portable to other systems (despite Rocker customization), please also check that the image I built scored on your machine.
* Switch to the branch `M1chaelM/starter_test`.
* Verify that the `team_config` folder contains an additional example called `2023_starter_test`.
* Follow the [testing tutorial](https://github.com/osrf/vrx/wiki/vrx_2023-testing) using
  ```
  TEAM=2023_starter_test
  TASK=stationkeeping
  TRIAL=0
  ```
The test should run normally and the boat should move a little bit after the task enters the running state. As long as it works properly, this branch can be discarded.

## Bonus
I updated the `component_config.yaml` for the `example_team` to fix the parameter `sensor_name` for the pinger. This can be tested by running `./prepare_team_wamv.bash example_team` and checking that it produces a `urdf`.

### CODE REVIEW

**Comment by M1chaelM** [comment]
On closer inspection of the tutorial and state of the repository, I'm converting this PR to a draft so I can address a few more things:
* [ ] It would be better to rename the image we have here to "simple" instead of starter, since it is closer to the image we called "simple" for VRX 2022.
* [ ] We should also add an analog to the VRX 2022 starter image with a readme explaining how to use it. The reason is to give an example of how to build a workspace and set up a development environment in the container, and this is a little more complicated.
* [ ] The [README](https://github.com/osrf/vrx-docker/blob/master/mwes/README.md) for the MWE directory as a whole needs to be updated.

**Review by caguero — CHANGES_REQUESTED** [review]
The instructions worked for me and I was able to run the solution and verify that the boat moves.

I just left a few minor comments.

**Review by caguero — CHANGES_REQUESTED** [review]
The instructions worked for me and I was able to run the solution and verify that the boat moves.

I just left a few minor comments.

File: mwes/2023/vrx_2023_starter/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Note: you must first build the `dockwater:humble` image locally as described
+# in the README
+ARG BASEIMG=dockwater:humble
+FROM $BASEIMG 
+
+# Set Ubuntu release
```
[the remark this answers is not in the document]
Comment (M1chaelM) [reply mwes/2023/vrx_2023_starter/Dockerfile:6]: This is done in the base image.

File: mwes/2023/vrx_2023_starter/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+#&& apt clean -qq
+
+# Optional: Uncomment and customize to add dev. tools, applications, etc. 
+#RUN apt-get update && apt-get install -y --no-install-recommends \
+#    gdb \
+#    psmisc \
+#    vim \
+#    && rm -rf /var/lib/apt/lists/*
+
+# Set up a catkin workspace
```
Comment (M1chaelM) [inline mwes/2023/vrx_2023_starter/Dockerfile:25]:
```suggestion
# Set up a colcon workspace
```

File: mwes/2023/vrx_2023_starter/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+#RUN apt-get update && apt-get install -y --no-install-recommends \
+#    gdb \
+#    psmisc \
+#    vim \
+#    && rm -rf /var/lib/apt/lists/*
+
+# Set up a colcon workspace
+RUN mkdir -p /vrx_ws/src
+
+# Here we use my_vrx as an example of a source code repository stored
```
Comment (caguero) [inline mwes/2023/vrx_2023_starter/Dockerfile:28]: `my_source` instead of `my_vrx`?

**Review by caguero — APPROVED** [review]

**Pull request merged**